Source/NSDictionaryController.m

```diff
@@ -0,0 +1,480 @@
+/* Implementation of class NSDictionaryController
+   Copyright (C) 2021 Free Software Foundation, Inc.
+
+   By: Gregory John Casamento
+   Date: 16-10-2021
+
+   This file is part of the GNUstep Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSException.h>
+#import <Foundation/NSIndexSet.h>
+#import <Foundation/NSKeyValueObserving.h>
+#import <Foundation/NSPropertyList.h>
+#import <Foundation/NSString.h>
+
+#import "AppKit/NSDictionaryController.h"
+#import "AppKit/NSKeyValueBinding.h"
+
+#import "GSBindingHelpers.h"
+#import "GSFastEnumeration.h"
+
+// Private methods for NSString and NSDictionary...
+@interface NSDictionary (GSPrivate_DictionaryToStrings)
+
+- (NSString *) _stringsFromDictionary;
+
+@end
+
+@implementation NSDictionary (GSPrivate_DictionaryToStrings)
+
+- (NSString *) _stringsFromDictionary
+{
+  NSEnumerator *en = [self keyEnumerator];
+  NSString *k = nil;
+  NSString *result = @"";
+
+  while ((k = [en nextObject]) != nil)
+    {
+      NSString *v = [self objectForKey: k];
+      result = [result stringByAppendingString:
```

Comment:
This format will only work for simple value types. Why not use one of the NSPropertyListSerialization formats?

Reply:
This shouldn't be an issue since it is only supposed to work for the strings format anyway.  I can find a function to convert from a strings list to a property list, but not the other way around. :/

Reply:
I will submit a quick fix for this if I find one or someone points it out.